### ISSUE TITLE
Decoder hang with invalid URL

### ISSUE THREAD

**Issue by elalish** [MONOGRID/gainmap-js#16]
We have a test that's now failing upon incorporating `HDRJPGLoader` in https://github.com/google/model-viewer/pull/4578:

```
test('throws on invalid URL', async () => {
      try {
        await textureUtils.loadEquirect('');
        expect(false).to.be.ok;
      } catch (e) {
        expect(true).to.be.ok;
      }
    });
```

It appears to time out reliably, so I think it's just hanging instead of throwing? Our similar test for a not found URL is throwing properly: `await textureUtils.loadEquirect('./nope.png');`

**Comment by elalish** [comment]
Also, it's interesting to note that `HDRJPGLoader` successfully loads not just SDR JPEG images, but also PNGs (which is great!). However, it makes the name slightly misleading...

**Comment by daniele-pelagatti** [comment]
I'm getting a proper error thrown when trying to load an invalid url in the three.js example: 

with both the sync method
![image](https://github.com/MONOGRID/gainmap-js/assets/19731017/126890bb-f193-488d-b3cf-75660af1e0bd)

and the async one
![image](https://github.com/MONOGRID/gainmap-js/assets/19731017/cb554feb-d188-47c5-93ae-e805af84d95f)

I need to investigate this further

**Comment by daniele-pelagatti** [comment]
ahhh, right, nevermind my previous comment, I misread your report, this seems to happens only with an empty string, I can reproduce the problem now, I'm investigating

**Comment by daniele-pelagatti** [comment]
this should be fixed in v3.0.0
